esphome/components/sensor/__init__.py

```diff
@@ -593,6 +594,29 @@ async def throttle_filter_to_code(config, filter_id):
     return cg.new_Pvariable(filter_id, config)
 
 
+TIMEOUT_WITH_PRIORITY_SCHEMA = cv.maybe_simple_value(
+    {
+        cv.Required(CONF_TIMEOUT): cv.positive_time_period_milliseconds,
+        cv.Optional(CONF_VALUE, default="nan"): cv.Any(
+            cv.templatable(cv.float_), [cv.templatable(cv.float_)]
+        ),
```

Comment:
I know you took this from existing code, but does this work?
```suggestion
        cv.Optional(CONF_VALUE, default="nan"): cv.ensure_list(
            cv.templatable(cv.float_)
        ),
```

Reply:
Why yes, yes it does...

Reply:
[`5081da5` (#9937)](https://github.com/esphome/esphome/pull/9937/commits/5081da5f19a960d7285e6b7e54e3528e577f9ce6)